internal/registration/internal/application/test/handlers_test.go

```diff
@@ -0,0 +1,97 @@
+package test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+
+	"GoEdu/internal/registration/internal/application"
+)
+
+func TestRegisterNewUser(t *testing.T) {
+	tests := map[string]struct {
+		input []application.RegisterNewUserCommand
+		want  bool
+	}{
+		"ok": {input: []application.RegisterNewUserCommand{
+			application.RegisterNewUserCommand{
+				FirstName: "Jon",
+				LastName:  "Jackson",
+				Email:     "jj@gmail.com",
+				Password:  "jj",
+			},
+		},
+			want: true,
+		},
+		"error": {input: []application.RegisterNewUserCommand{
+			application.RegisterNewUserCommand{
+				FirstName: "Jon",
+				LastName:  "Jackson",
+				Email:     "jj@gmail.com",
+				Password:  "jj",
+			},
+			application.RegisterNewUserCommand{
+				FirstName: "Jon",
+				LastName:  "Jackson",
+				Email:     "jj@gmail.com",
+				Password:  "jj",
+			},
+		},
+			want: false,
+		},
+	}
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			repo := NewMockRepository()
```

Comment:
For now you shouldn't provide any working implementation for repositories, email verifiers, password hashers. For each test provide similar but different test double. At this point we should check only calling this dependencies rather to wait correct behavior of those.